subprojects/gradle/build-verdict/src/main/kotlin/com/avito/android/build_verdict/UserDefinedTaskVerdictProducer.kt

```diff
@@ -0,0 +1,28 @@
+package com.avito.android.build_verdict
+
+import org.gradle.api.Task
+
+class UserDefinedTaskVerdictProducer(
+    predicate: TaskPredicate,
+    producer: TaskVerdictProducer
+) : TaskPredicate by predicate, TaskVerdictProducer by producer
+
+interface TaskPredicate {
+    fun accept(task: Task): Boolean
+
+    class ByName(private val name: String) : TaskPredicate {
+        override fun accept(task: Task) = task.name == name
+    }
+
+    class ByClass(private val acceptedClass: Class<in Task>) : TaskPredicate {
+        override fun accept(task: Task) = acceptedClass.isInstance(task)
+    }
+}
+
+interface TaskVerdictProducer {
+    fun produce(task: Task): String
+
+    class Lambda(private val producer: (Task) -> String) : TaskVerdictProducer {
```

Comment:
I see no point in this additional class, doesn't 

```
companion object {
   internal fun create(producer: (Task) -> String) = object : TaskVerdictProducer {
      override fun produce(task: Task) = producer(task)
   }
}
```

tells intention the easier way?
I see this lambda as some synthetic thing